### PR TITLE
The Witness: New solution to that plando issue

### PR DESCRIPTION
**Pull request by NewSoupVi**
## History

The Witness wants to do one simple thing.
"Pre-place a random symbol item onto the first check in the game, so that the player has things to do locally and sphere 1 has a bit of variety."
It should be simple, right? Just fill the item during `create_items` or something.
Well, it would be simple. If it weren't for plando.

Here's the problem. If a player uses `plando_items` with one of the eligible symbol items and uses `from_pool: true`, now their generation will inexplicably crash some of the time. Reason: That symbol item might already be gone from the itempool.

## The current solution

We first check whether the item will be plandoed before placing it, before plando ever "resolves".
This is terrible, because people can have *random* plando. Also, plando blocks can take many different shapes.
Look at the code removed in `items.py`, then think about the fact that *other worlds are starting to copy that code*.

## The new solution

Place this item during `fill_hook`. This basically uses the current plando code, but modified.

### CODE REVIEW

**Comment by NewSoupVi** [comment]
Changed this to using pre_fill because I learned that fill_hook is after distribute_early_items

**Comment by NewSoupVi** [comment]
Drafting this for two reasons:

- I wanna add unit tests for this.
- I want to check for `location.excluded`.

File: worlds/witness/__init__.py [hunk 1]
```diff
@@ -304,6 +287,79 @@ def create_items(self) -> None:
...
+                  fill_locations: List[Location]) -> None:
+        if not self.options.early_symbol_item:
+            return
+        if self.options.puzzle_randomization == "sigma_expert":
+            return
+        # Non-Expert: Pick an early item to put on Tutorial Gate Open.
+        # Done after plando to avoid conflicting with it.
+        # Done in fill_hook because multiworld itempool manipulation is not allowed in pre_fill.
+
+        player_name = self.multiworld.get_player_name(self.player)
```
Comment (ScipioWright) [inline worlds/witness/__init__.py:347]:
```suggestion
        player_name = self.player_name
```

**Comment by NewSoupVi** [comment]
Undrafting because I've added a unit test and made it more robust in terms of things like excluded locations.

**Pull request closed without merging**